### ISSUE TITLE
authentication page fails to open in firefox

### ISSUE THREAD

**Issue by immanuelabosh** [yuezk/GlobalProtect-openconnect#407]
**Describe the bug**
When using --default-browser with firefox as the default browser if firefox is already open then I get the following error 

`Firefox is already running, but is not responding. To use Firefox, you must first close the existing Firefox process, restart your device, or use a different profile.`

If I set edge as the default browser no error occurs. If I use xdg-open to open a url in firefox it open fine. If firefox isn't open it correctly launches firefox and any subsequent attempts correctly open a new tab in firefox. 

**Expected behavior**
If firefox is already open a new tab should open in firefox.

**Screenshots**
![image](https://github.com/user-attachments/assets/42aaa9d5-88a1-42f6-af69-8eed3c90ee36)

**Logs**
##@#####:~$ sudo gpclient connect ###.#####.### -g #### --default-browser
Place your right index finger on the fingerprint reader
Failed to match fingerprint
Place your right index finger on the fingerprint reader
Failed to match fingerprint
Place your right index finger on the fingerprint reader
Place your finger on the reader again
[2024-08-07T23:13:10Z INFO  gpclient::cli] gpclient started: 2.3.4 (2024-07-09)
[2024-08-07T23:13:10Z INFO  gpapi::portal::prelogin] Portal prelogin with user_agent: PAN GlobalProtect
[2024-08-07T23:13:10Z INFO  gpauth::cli] gpauth started: 2.3.4 (2024-07-09)
[2024-08-07T23:13:10Z INFO  gpauth::cli] Please continue the authentication process in the default browser
[2024-08-07T23:13:10Z INFO  gpclient::connect] Waiting for the browser authentication to complete...
[2024-08-07T23:13:10Z INFO  gpclient::connect] Listening authentication data on port 43303

**Environment:**
 - OS: [Fedora 40]
 - Desktop Environment: [KDE]
 - Is remote SSH? [No]

**Additional context**

**Comment by immanuelabosh** [comment]
Please let me know if there's any other logs or troubleshooting steps you'd like me to do.

**Comment by yuezk** [comment]
Thanks for the feedback. I will try to reproduce it with Firefox.

**Comment by yuezk** [comment]
@immanuelabosh I can reproduce this problem. You can try to use `sudo -E …` to see if it works.

**Comment by immanuelabosh** [comment]
Can confirm this works, thank you

**Comment by leinardi** [comment]
Hi @yuezk , thank you for the suggestion to try `sudo -E`. I gave it a shot, but unfortunately, it didn't resolve the issue. In fact, when I use the `-E` parameter, Firefox does not open at all. Below is the output from the CLI client:
```
$ sudo -E gpclient --fix-openssl connect redacted-address.com --default-browser
[2024-08-28T13:22:06Z INFO  gpclient::cli] gpclient started: 2.3.7 (2024-08-16)
[2024-08-28T13:22:06Z INFO  gpapi::portal::prelogin] Portal prelogin with user_agent: PAN GlobalProtect
[2024-08-28T13:22:06Z INFO  gpapi::portal::prelogin] Perform prelogin, user_agent: PAN GlobalProtect
[2024-08-28T13:22:06Z INFO  gpauth::cli] gpauth started: 2.3.7 (2024-08-16)
[2024-08-28T13:22:06Z INFO  gpauth::cli] Fixing OpenSSL environment
[2024-08-28T13:22:06Z INFO  gpapi::process::browser_authenticator] Launching the default browser...
[2024-08-28T13:22:06Z INFO  gpauth::cli] Please continue the authentication process in the default browser
[2024-08-28T13:22:06Z INFO  gpauth::cli] Listening authentication data on port 38247
[2024-08-28T13:22:06Z INFO  gpauth::cli] If it hangs, please check the logs at `/tmp/gpcallback.log` for more information
```

Additionally, no log file is present at `/tmp/gpcallback.log` as indicated in the output. I'm still experiencing the same issue where Firefox does not open correctly, and now with `sudo -E`, it fails to open altogether. Any further guidance on how to resolve this would be greatly appreciated.

**Comment by yuezk** [comment]
Consider using the following command instead because running GUI with sudo
could have some unexpected problems.

`gpauth <portal> --browser default 2>/dev/null | sudo gpclient connect
<portal> --cookie-on-stdin`

**Comment by leinardi** [comment]
Hi @yuezk, thank you for your answer. I did some more tests today and I found out this:

These 2 commands both work fine and open Firefox correctly:
```bash
sudo -E gpclient --fix-openssl connect redacted-address.com --browser firefox
gpauth redacted-address.com --browser firefox 2>/dev/null | sudo gpclient connect redacted-address.com --cookie-on-stdin
```

but if I replace `--browser firefox` with `--browser default` or `--default-browser` Firefox, the default browser, is not open but `telegram-desktop` is opened instead! I think it's an issue with the detection of the default browser so I will add more details in  #416.
